### PR TITLE
[CF-120] Fix "Port is still in use" when attempt boot VM second time

### PR DESCRIPTION
**Pull request by antonf**
Fix an issue when we trying boot VM in the same availability zone,
but on another host. For example we have this case when the VM failed
during deployment on the first host. Before this patch we had exception
"Port is still in use", because port is still attached to the failed VM.
With this patch CloudFerry recreate ports for the VM and trying boot it
on another host.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:124: 
[E1120(no-value-for-parameter), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] No value 
for argument 'availability_zone' in method call
tests/cloudferrylib/os/actions/test_transport_instance.py:124: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3210/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:125: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:125: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by yosharma** [comment]
+1, Tested in staging environment

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:125: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:125: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:125: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/transport_instance.py [hunk 1]
```diff
@@ -127,26 +134,90 @@ def get_dst_image(self, src_image_id):
...
+
+        _instance = one_instance[utils.INSTANCES_TYPE].values()[0]
+        instance_body = _instance[utils.INSTANCE_BODY]
+        hosts = dst_compute.get_compute_hosts(availability_zone)
+        if not hosts:
+            message = ("No hosts in availability zone '{az}' "
+                       "found on destination.").format(az=availability_zone)
+            raise DestinationCloudNotOperational(message)
+
+        random.seed()
```
Comment (SVilgelm) [inline cloudferrylib/os/actions/transport_instance.py]: `random.seed()` is not necessary.
[seed documentation](https://docs.python.org/2/library/random.html#random.seed)
> current system time is also used to initialize the generator when the module is first imported.

**Comment by SVilgelm** [comment]
:+1: can be merged

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_transport_instance
tests/cloudferrylib/os/actions/test_transport_instance.py:64: [W0212(protected-access), 
DeployInstanceWithManualScheduling.test_tries_to_boot_vm_on_all_nodes] Access to a 
protected member _deploy_instance_on_random_host of a client class
tests/cloudferrylib/os/actions/test_transport_instance.py:125: 
[W0212(protected-access), 
DeployInstanceWithManualScheduling.test_runs_only_one_boot_if_node_is_good] Access to a 
protected member _deploy_instance_on_random_host of a client class
************* Module cloudferrylib.os.compute.nova_compute
cloudferrylib/os/compute/nova_compute.py:21: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
cloudferrylib/os/compute/nova_compute.py:816: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:18: [E0611(no-name-in-module), ] No name 
'client' in module 'novaclient.v1_1'
tests/cloudferrylib/os/compute/test_nova.py:415: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:418: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:420: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:448: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:453: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:483: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:484: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:488: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
Didn't work

**Pull request merged**